File: 07.py
import math
numeral = input("Ingresa un número: ")


def isNar(numeral):
    nar = []
    for n in numeral:
        n = int(n)
        nar.append(int(math.pow(n, len(numeral))))
        nar_sum = sum(nar)
    if nar_sum == int(numeral):
        print(f"{numeral} es un número narcisista.")
    else:
        print(f"{numeral} no es un número narcisista.")


isNar(numeral)

File: 07.js
let numeral = prompt('Ingresa un número');

function isNar(num) {
  let nar = [];
  for (let n in num) {
    base = num[n];
    exponente = num.length;
    rta = Math.pow(base, exponente);
    nar.push(rta);
  }
  const total = nar.reduce((a, b) => a + b, 0);

  if (Number(total) === Number(num)) {
    console.log(`El ${numeral} es un número narcisista`);
  } else {
    console.log(`El ${numeral} No es un número narcisista`);
  }
}

isNar(numeral);
